pkg/client/cache/delta_fifo.go

```diff
@@ -404,6 +416,16 @@ func (f *DeltaFIFO) Pop(process PopProcessFunc) (interface{}, error) {
 	defer f.lock.Unlock()
 	for {
 		for len(f.queue) == 0 {
+			// When the queue is empty, invocation of Pop() is blocked until new item is enqueued.
```

Comment:
The entire PR is about getting out of the control loop when the queue is empty. It does not cover situations where the queue is not empty. I am leaving the non-empty queue case with all tests to be covered in different PR.